### PR TITLE
time period 

### PR DESCRIPTION
**Pull request by marqh**
time period interpretation for name files

### CODE REVIEW

File: lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# (C) British Crown Copyright 2013 - 2015, Met Office
```
Comment (dkillick) [inline lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py]: Well that's definitely not right...

File: lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import datetime
+
+from iris.fileformats.name_loaders import _calc_integration_period
+
+
+class Tests(tests.IrisTest):
```
Comment (dkillick) [inline lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py]: Typically we'd call this `Test`.

**Comment by marqh** [comment]
these are just the tests, I'm expecting failures, a follow up PR is inbound

please do not merge at 0f84055d19b5adead051148fd13874e42f3a3a27

**Comment by marqh** [comment]
minor review changes, hopefully merge suitable now

File: lib/iris/fileformats/name_loaders.py [hunk 3]
```diff
@@ -193,19 +193,22 @@ def _calc_integration_period(time_avgs):
     """
     integration_periods = []
     pattern = re.compile(
-        r'(\d{0,2})(day)?\s*(\d{1,2})(hr)?\s*(\d{1,2})(min)?\s*(\w*)')
+        r'\s*(\d{1,2}day)?\s*(\d{1,2}hr)?\s*(\d{1,2}min)?\s*(\w*)\s*')
```
Comment (dkillick) [inline lib/iris/fileformats/name_loaders.py]: The regex allows for any number of trailing whitespace characters at the end of the time string, but this behaviour is not tested. You could add a test that just confirms that the pattern is still appropriately matched with trailing whitespace.

File: lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        self.assertEqual(result, expected)
+
+    def test_3_hour_av(self):
+        time_avgs = ['          3hr 0min average']
+        result = _calc_integration_period(time_avgs)
+        expected = [datetime.timedelta(0, (3*60*60))]
+        self.assertEqual(result, expected)
+
+    def test_3_hour_int(self):
+        time_avgs = ['         3hr 0min integral']
```
Comment (dkillick) [inline lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py]: I see that a time average can be either an `average` or an `integral` but that the behaviour of Iris in either case is the same - does the type of time average make no further difference to the interpretation of the time average?

Comment (marqh) [inline lib/iris/tests/unit/fileformats/name_loaders/test__calc_integration_period.py]: no difference, that is handled in a different part of the code

**Comment by dkillick** [comment]
Thanks @marqh!

**Comment by SusanJL** [comment]
This looks like it fixes the problem. Thank you.
